### ISSUE TITLE
PostCode table procedure LookupPostCode parameters length

### ISSUE THREAD

**Issue by BertDeTemmerman** [microsoft/ALAppExtensions#14298]
Why are the parameters of `procedure LookupPostCode(var CityTxt: Text; var PostCode: Code[20]; var CountyTxt: Text; var CountryCode: Code[10])` on table 225 "Post Code" (unlimited) text? And not the actual length of the fields?

Further in the procedure these parameters are assigned 
```
            PostCode := PostCodeRec.Code;
            CityTxt := PostCodeRec.City;
            CountryCode := PostCodeRec."Country/Region Code";
            CountyTxt := PostCodeRec.County;
```
where
```
field(2; City; Text[30])
field(5; County; Text[30])
```

Which means `CityTxt` and `CountyTxt` can never be more than the original field length of 30.

Calling this function with own fields (of the correct size), results in warning AA0139:
```
PostCode.LookupPostCode("From City", "From Post Code", "From County", "From Country/Region Code");
Possible overflow assigning 'Text' to 'Text[30]'.AL(AA0139)
```

Checked in Microsoft_Base Application_18.4.28601.29407

Edit:
`procedure ValidateCity(var CityTxt: Text[30]; var PostCode: Code[20]; var CountyTxt: Text[30]; var CountryCode: Code[10]; UseDialog: Boolean)` uses the correct lengths

**Comment by StanPesotskiy** [comment]
The functions with long Text parameters for City and County was added 3 years ago and at that time we got some warnings saying that this function is called with parameters longer than 30 chars. The function got Text params.
Now as it is a public function we cannot simply change Text to Text[30] as it may break some partner's code.
The only solution is to make another function with Text[30] params and replace all or most calls of old functions. But the only benefit would be a reduced number of false warnings, and they are false because don't lead to an actual overflow unless someone overwrites the standard code using the event OnBeforeLookupPostCode(). 
The risk of breaking someone and the small benefit do not worth such a change like replacing all calls of LookupPostCode() and CheckClearPostCodeCityCounty().
